Lb testcases

KunTares opened the pull request:
Added testcases lb.py, lb_gpu.py and lb_stokes_sphere.py.
Partial fix for lbboundaries force interface.

KunTares committed:
partial fix for lbboundaries force interface

fweik committed:
lbboundaries: Bind get_force in python if.

fweik committed:
lbboundaries: Fix type for float params

KunTares committed:
Added lb_stokes_sphere testcase.

KunTares committed:
Added testcases lb.py and lb_gpu.py

rempferg commented:
The test case lb.py requires a file lb_system.data, which seems to be missing.

rempferg commented on testsuite/python/lb.py:
```diff
@@ -0,0 +1,163 @@
+			del system.actors[0]
+		system.part.clear()
+		#import particle data
+		data = np.genfromtxt("lb_system.data")
```
Please add this file.

rempferg committed:
Corrected CMakeLists.txt for python test cases and added missing entries.

rempferg committed:
Removed erroneous integrate.integrate(100) lines from lb_test*.py

rempferg reviewed (changes requested):
- I added all existing test cases to make check and corrected that one erroneous comma
- I removed erroneous integrate.integrate(100) lines from lb_test*.py
- I added all existing test cases to make check and corrected that one erroneous comma
- added ut decorators to only run minimize_energy test if LJ is compiled in

TODO
- add lb_system.data
- lb_test*.py do not contain asserts
- add decorators to only run LB tests of respective LB features compiled in

rempferg committed:
Added ut decorators to only run minimize_energy.py test if LJ is compiled in.

KunTares committed:
Added missing file lb_system.data

KunTares committed:
Removed unnecessary dependence on ELECTROSTATICS from lb.py and lb_GPU.py
Added decorators to only run lb.py and lb_GPU.py tests if those features are compiled in

KunTares committed:
Removed unnecessary tests lb_test*.py

KunTares committed:
Changed skip condition to cover all needed features of lb*.py and lb_stokes_sphere.py
Fixed lb_stokes_sphere.py using GPU

KunTares committed:
Removed lb_stokes_sphere.py

hmenke commented on testsuite/python/lb.py:
```diff
@@ -12,6 +12,10 @@
 
 
+
+@ut.skipIf('LB' not in espressomd.code_info.features(),
```
Please use `has_features`.  The `code_info` is not a user module.
````python
@ut.skipIf(not espressomd.has_features(["LB"]),
           "Features not available, skipping test!")
````

hmenke commented on testsuite/python/lb_gpu.py:
```diff
@@ -11,7 +11,8 @@
 import numpy as np
 
-
+@ut.skipIf('LB_GPU' not in espressomd.code_info.features(),
```
Please use `has_features`.  The `code_info` is not a user module.
````python
@ut.skipIf(not espressomd.has_features(["LB_GPU"]),
           "Features not available, skipping test!")
````

hmenke commented on testsuite/python/lb.py:
```diff
@@ -56,7 +60,7 @@ def test(self):
 			q = particle[2]
```
`q` is now unused.  Please remove.

hmenke commented on testsuite/python/lb_gpu.py:
```diff
@@ -56,7 +57,7 @@ def test(self):
 			q = particle[2]
```
`q` is now unused. Please remove.

KunTares committed:
Replacing code_info.features with has_features

KunTares committed:
Removed unsused variable q from lb*.py

KunTares committed:
Removing unneeded tests lb_test*.py and lb_stokes_sphere.py from CMakeLists.txt

rempferg committed:
Merge branch 'python' into lb_testcase

rempferg commented:
I don't understand how the particle temperature can be 2 here. The calculation seems correct to me. Also, I can't reproduce this locally.

fweik committed:
Merge remote-tracking branch 'upstream/python' into HEAD

fweik committed:
core: Fixed regression in lb parameter distribution.

fweik committed:
Merge remote-tracking branch 'upstream/python' into lb_testcase

fweik committed:
testsuite: abspath...

fweik committed:
Merge branch 'python' of github.com:espressomd/espresso into lb_testcase

fweik committed:
Merge branch 'python' of github.com:espressomd/espresso into lb_testcase

fweik committed:
Merge remote-tracking branch 'origin/gitlab' into lb_testcase

fweik committed:
python: Corrected initialization order for lb.

fweik committed:
testsuite: Made pathes in engine_* tests abs

fweik committed:
lb: Correct initialization order for lb gpu.

fweik committed:
testsuite: Added and fixed lb, lb_gpu and lb_stokes_sphere_gpu tests

fweik commented:
@rempferg @KunTares please check again. If fixed the two thermostat tests, and add the gpu stokes sphere test, which seemed ok.

fweik committed:
Merge branch 'python' into lb_testcase

fweik committed:
core: make lattice_switch xor

fweik committed:
Merge branch 'lb_testcase' of https://github.com/KunTares/espresso into lb_testcase

rempferg commented:
Looks good to me. Unfortunately I don't see a way to make the calculation of the fluid statistics faster without significant work.
